wazo_dird_plugin_odoo/plugin.py

```diff
@@ -19,89 +19,39 @@ class OdooView(BaseBackendView):
     list_resource = http.OdooList
     item_resource = http.OdooItem
 
+class OdooPlugin(BaseServicePlugin):
 
-class OdooPlugin(BaseSourcePlugin):
+    def name(self):
+        return 'Odoo backend'
 
-    def load(self, dependencies):
-        config = dependencies['config']
+    def load(self, config):
+        self.db = config['config']['odoo_config']['database']
+        self.server = config['config']['odoo_config']['server']
+        self.username = config['config']['odoo_config']['username']
+        self.password = config['config']['odoo_config']['password']
+        self.model = config['config']['odoo_config']['model']
 
-        self.name = config['name']
-        self.sock = xmlrpclib.ServerProxy(
-            'http://%s:%s/xmlrpc/object' % (
-                config['server'],
-                config['port']))
-        self.db = config['database']
-        self.uid = config['userid']
-        self.pwd = config['password']
+        logger.debug('Starting Odoo authentication')
+        common = xmlrpc.client.ServerProxy('{}/xmlrpc/2/common'.format(self.server))
+        self.uid = common.authenticate(self.db, self.username, self.password, {})
+        logger.info('Authentication succeeded')
+        logger.info('Backend loaded')
 
-        unique_column = None
-        format_columns = dependencies['config'].get(self.FORMAT_COLUMNS, {})
-        if 'reverse' not in format_columns:
-            logger.info(
-                'no "reverse" column has been configured on %s will use "givenName"',
-                self.name
-            )
-            format_columns['reverse'] = '{givenName}'
-
-
-        self._SourceResult = make_result_class(
```

Comment:
Is all this part essential in the code? From my understanding, this has to do with formatting in output, am I correct? Can I put aside this part or will it create some problems with the output's handling?